### DISCUSSION TITLE
Applying SpecAug in ASR training

### DISCUSSION THREAD

**Discussion by Bereket-Desbele** [espnet/espnet#5248] [Q&A]
I have used SpecAug previously before to enhance the performance of a transformer based ASR (on ESPnet 2 version 202207) and has worked well. Now I have tried to repeat the same task on ESPnet 2 version 202301 but for some reason the training isn't converging at all. Is anyone having the same problem??

N.B. I haven't changed any parameters in both scenarios. But the results in ESPnet 2 version 202301 don't make sense. The validation results are much higher than the training results, and if I am not mistaken it's supposed to be the other way around.

**Comment by Bereket-Desbele** [comment]
With validation results I meant validation loss is mich higher than training loss and validation accuracy is much lower than training accuracy. But it's supposed to the other way around.

**Reply by sw005320** [reply]
Can you paste the learning curves of both versions, if possible?

**Reply by Bereket-Desbele** [reply]
I couldn't find learning curve for the (ESPnet 2 version 202301) so I have just run & got the following results. But the training re not complete

![optim0_lr0](https://github.com/espnet/espnet/assets/41731956/1976b844-fcf5-46c8-b0d7-9136f0ec7908)
This is the original with lr 0.001 & 2500steps, 64 batch size (fold) & specaug.(ESPnet 2 version 202207)

![Uploading optim0_lr0 - batch_size_32.png…]()
This is the same as the original but with 32 batch size(ESPnet 2 version 202301)

![Uploading optim0_lr0 - acc_grad_8.png…]()
& This is the same as the 2nd one but with added accumulated gradient of 8.(ESPnet 2 version 202301)

**Reply by Bereket-Desbele** [reply]
![optim0_lr0 - batch_size_32](https://github.com/espnet/espnet/assets/41731956/82dc87a0-af7d-4f46-90f8-975ce0f5a53a)
This is the original with lr 0.001 & 2500steps, 64 batch size (fold) & specaug.(ESPnet 2 version 202207)

![optim0_lr0 - acc_grad_8](https://github.com/espnet/espnet/assets/41731956/c1df137d-3677-4b14-b457-298c6785f4cb)
This is the same as the original but with 32 batch size(ESPnet 2 version 202301)

**Reply by Bereket-Desbele** [reply]
Sorry but The captions are wrong. The pictures are in the following order

The first is the original with lr 0.001 & 2500steps, 64 batch size (fold) & specaug.(ESPnet 2 version 202207)

The second is the same as the original but with 32 batch size(ESPnet 2 version 202301)

& The third is the same as the 2nd one but with added accumulated gradient of 8.(ESPnet 2 version 202301)
